go/mysql/conn.go

```diff
@@ -485,14 +492,9 @@ func (c *Conn) writeEphemeralPacket(direct bool) error {
 		}
 	case ephemeralWriteBigBuffer:
 		// This is the slower path for big data.
-		// With direct=true, the caller expects a flush, so we call it
-		// manually.
-		if err := c.writePacket(*c.currentEphemeralWriteBuffer); err != nil {
+		if err := c.writePacket(w, *c.currentEphemeralWriteBuffer); err != nil {
```

Comment:
perhaps make this not a function on `conn` anymore?

Reply:
ah. we need sequence number. that said, if we take suggestion above of swapping out writer at a higher level (when we start buffering), then we no longer need to pass the writer in as an argument to `writePacket`